blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/astyanax/AstyanaxWriter.java

```diff
@@ -281,6 +327,10 @@ public void insertMetrics(Collection<IMetric> metrics, ColumnFamily cf) throws C
                             Instrumentation.markFullResPreaggregatedMetricWritten();
                         }
                     }
+
+                    if (locatorInsertOk) {
+                        insertDelayedLocator(metric, batch, isTrackingDelayedMetrics, clock);
```

Comment:
We only call the regular insertLocator() only if it hasn't been inserted before. At least I think that's what the cache that's used in isLocatorCurrent() is for. Do we want to do the same here for delayed locator?